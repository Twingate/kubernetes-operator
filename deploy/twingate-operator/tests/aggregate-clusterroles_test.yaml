suite: Aggregate Cluster Roles
templates:
  - aggregate-clusterroles.yaml
values:
  - values.test.yaml
tests:
  - it: should enable aggregate cluster roles
    set:
      rbac:
        createAggregateClusterRoles: true
    asserts:
      - matchSnapshot: {}
