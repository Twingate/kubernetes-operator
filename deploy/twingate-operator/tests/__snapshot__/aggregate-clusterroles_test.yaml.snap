should enable aggregate cluster roles:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
        rbac.authorization.k8s.io/aggregate-to-admin: "true"
        rbac.authorization.k8s.io/aggregate-to-edit: "true"
        rbac.authorization.k8s.io/aggregate-to-view: "true"
      name: RELEASE-NAME-twingate-operator-crd-view
    rules:
      - apiGroups:
          - twingate.com
        resources:
          - '*'
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
        rbac.authorization.k8s.io/aggregate-to-admin: "true"
        rbac.authorization.k8s.io/aggregate-to-edit: "true"
      name: RELEASE-NAME-twingate-operator-crd-edit
    rules:
      - apiGroups:
          - twingate.com
        resources:
          - '*'
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
